src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -165,7 +165,12 @@ const HeatmapPlot = (props) => {
   }, [louvainClusterCount]);
 
   useEffect(() => {
-    dispatch(updateCellInfo({ cellName: cellHighlight }));
+    if (cellHighlight) {
+      const containingClusters = getCellClassProperties(parseInt(cellHighlight, 10), ['louvain', 'scratchpad'], cellSets)
+        .map(({ name, rootClusterName }) => `${rootClusterName} : ${name}`);
+
+      dispatch(updateCellInfo({ cellName: cellHighlight, cellSets: containingClusters }));
```

Comment:
I am not sure I agree about this, I feel like the only information we need to share across components inside redux is which cell was highlighted, then each component can decide exactly what to display on their own tooltips.

For example in the embedding, when gene expressions for one particular gene is activated, it displays gene expression and other stuff, I feel like that doesn't need to be in redux, it can simply be stored in a `useRef` and recalculated when cellHighlight is updated.

I would probably not store `cellInfo` in redux, instead perhaps only `highlightedCell` or something like that.

Reply:
But when we hover on the heatmap we want the tooltip to appear on the embedding too so we need that cellInfo state to be checked in both components? How else would we update it?

Reply:
We still need to save cellSets inside - when we hover on the heatmap we want the tooltip to appear on the embedding with the cluster information too. What are the drawbacks with the current version? You think its better to add more variables in the files and more logic instead of keeping everything in redux?

Reply:
Each tooltip gets the things it needs from redux, ignoring the rest just like in many places accross the platform, I don't see the problem here, maybe we can discuss?

Reply:
> You think its better to add more variables in the files and more logic instead of keeping everything in redux?

The amount of logic is the same because we still need to calculate the same things, if it is well abstracted it's just a matter of having that code in the embedding instead of when dispatching to redux.


> Each tooltip gets the things it needs from redux, ignoring the rest just like in many places across the platform (getting only hierarchy under cellSets and etc.), I don't see the problem here, maybe we can discuss?

I would argue this isn't the same case as hierarchy, but we can dive into that later if you want.

I would encourage you to read these 2 points:
https://redux.js.org/style-guide/style-guide#keep-state-minimal-and-derive-additional-values
https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters

Apart from that, here I wrote down a more specific explanation on why I feel this pattern is better, regardless of whether redux suggests this or not.

**1:** The version where all the data each component needs gets sent to redux means that the following happens:
when cellId is updated in one of the components
- on dispatch that component needs to calculate all of the extra data for each of the components which **might or might not be subscribed to the cellId updates**, so that logic is actually being executed whenever cellId changes.

**2:** The version where we only send to redux what we absolutely need:
- on dispatch the component sends [only the event specific data](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters): the cellId
- Each component listens to the cellId changing and processes that in whatever way they need to

The differences:
- With case **1** there is an unnecessary coupling between redux and each component that uses it, coupling is bad because:
    - if a component no longer needs something we need to adapt redux and that component and perhaps other components too, depending on how the code for calculating that outdated data that is dispatched works.
    - If a component is removed we need to adapt redux and the other components for similar reasons, while inthe other way we simply remove the component and be happy.

- With case **1**, if a component was closed by the user due to the remove panels functionality we have in data exploration, processing that data for display in the closed component is still going to be done even though we don't need to in reality.

- I think there are other reasons, but I think the redux style guide is pretty useful if you want more information too.